### ISSUE TITLE
Rename TextPlainFormatter to string formatter

### ISSUE THREAD

**Issue by yishaigalatzer** [aspnet/Mvc#1631]
Changed - to renaming and cleanup from the suggestion below:

The following line should check for the `response.ContentType != null` before assignment.

https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc.Core/Formatters/OutputFormatter.cs#L193

@harshgMSFT @davidfowl  @lodejard this will enable returning a string but specifying a custom content type.

e.g. pseudo code.

```c#
public string MyAction()
{
    var d = GetData();
    var jsonString = Json.Serialize(d);
    jsonString = Manipulate(jsonString);

    Response.ContentType = "application/json";
    return jsonString;
}
```

or

```c#
public async Task<string> MyAction()
{
    var jsonString = await client.GetData("http://url");
    Response.ContentType = "application/json";
    return Manipulate(jsonString);
}
```

**Comment by yishaigalatzer** [comment]
Decided to leave this alone. It causes very interesting subtle issues with respecting CharSet. And ultimately the user can just return a ContentResult if he wants to customize the ContentType

**Comment by yishaigalatzer** [comment]
#1826 

Resolved by 9ac37fbc7
